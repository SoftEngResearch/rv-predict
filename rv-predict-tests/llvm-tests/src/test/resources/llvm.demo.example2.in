<locId:1;fn:__libc_start_main;file:/build/buildd/glibc-2.21/csu/libc-start.c;line:289>
<gid:0;tid:1;id:1;addr:0;value:0;type:INVOKE_METHOD>
<locId:2;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:18>
<gid:1;tid:1;id:2;addr:0;value:2;type:START>
<locId:3;fn:__tsan_thread_start_func;file:/home/traian/Documente/rv/rv-predict-llvm/projects/compiler-rt/lib/tsan/rtl/tsan_interceptors.cc;line:849>
<gid:2;tid:2;id:3;addr:0;value:0;type:INVOKE_METHOD>
<locId:4;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:19>
<gid:3;tid:1;id:4;addr:0;value:3;type:START>
<locId:5;fn:MyThread;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:9>
<varId:4;desc:global 'lock' of size 40 at 0x0000014a6828 (example2+0x0000014a6828)>
<gid:4;tid:2;id:5;addr:4;value:0;type:WRITE_LOCK>
<gid:5;tid:3;id:3;addr:0;value:0;type:INVOKE_METHOD>
<locId:6;fn:MyThread;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:10>
<varId:6;desc:global 'x' of size 4 at 0x0000014a6820 (example2+0x0000014a6820)>
<gid:6;tid:2;id:6;addr:4294967290;value:0;type:READ>
<locId:7;fn:MyThread;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:10>
<gid:7;tid:2;id:7;addr:4294967290;value:1;type:WRITE>
<locId:8;fn:MyThread;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:11>
<gid:8;tid:2;id:8;addr:4;value:0;type:WRITE_UNLOCK>
<locId:9;fn:MyThread;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:12>
<gid:9;tid:2;id:9;addr:4294967290;value:1;type:READ>
<locId:10;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:20>
<gid:10;tid:1;id:10;addr:4;value:0;type:WRITE_LOCK>
<gid:11;tid:2;id:3;addr:0;value:0;type:FINISH_METHOD>
<locId:11;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:21>
<gid:12;tid:1;id:11;addr:4294967290;value:1;type:READ>
<locId:12;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:21>
<gid:13;tid:1;id:12;addr:4294967290;value:2;type:WRITE>
<locId:13;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:22>
<gid:14;tid:1;id:13;addr:4;value:0;type:WRITE_UNLOCK>
<gid:16;tid:3;id:5;addr:4;value:0;type:WRITE_LOCK>
<gid:17;tid:3;id:6;addr:4294967290;value:2;type:READ>
<gid:18;tid:3;id:7;addr:4294967290;value:3;type:WRITE>
<gid:19;tid:3;id:8;addr:4;value:0;type:WRITE_UNLOCK>
<locId:14;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:23>
<gid:20;tid:3;id:9;addr:4294967290;value:3;type:READ>
<gid:15;tid:1;id:14;addr:4294967290;value:0;type:WRITE>
<locId:15;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:24>
<gid:22;tid:3;id:3;addr:0;value:0;type:FINISH_METHOD>
<gid:21;tid:1;id:15;addr:9;value:140146553911040;type:READ>
<locId:16;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:24>
<gid:23;tid:1;id:16;addr:0;value:2;type:JOIN>
<locId:17;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:25>
<gid:24;tid:1;id:17;addr:11;value:140146545518336;type:READ>
<locId:18;fn:main;file:/home/traian/Documente/rv/racey-c-programs/demo/example2.c;line:25>
<gid:25;tid:1;id:18;addr:0;value:3;type:JOIN>
<gid:26;tid:1;id:1;addr:0;value:0;type:FINISH_METHOD>
