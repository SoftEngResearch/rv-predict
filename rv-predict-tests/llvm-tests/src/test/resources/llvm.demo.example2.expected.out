Data race on global 'x' of size 4 at 0x0000014a0bd8 \(a\.out \+ 0x0000014a0bd8\): \{\{\{(.*)---->  at fn:MyThread;file:<null>;line:0(.*)---->  at fn:main;file:<null>;line:0(.*)\}\}\}
